docs/rfcs/0002-storing-course-data.md

```diff
@@ -0,0 +1,96 @@
+  after we switch to using our own Open edX instance.
+
+Out of scope for this RFC:
+- Content management – we will just rely on Django admin for creating/updating
+  courses by hand.
+- Any kind of grading/certificate data.
+
+\* - _Much like MicroMasters, we will keep our own models for courses/programs
+  that will exist in parallel with the course data in edX_
+
+### Models
+
+Since this app has very similar needs, `Program`, `Course`, and `CourseRun` 
+models will very closely mirror the [models in MicroMasters](https://github.com/mitodl/micromasters/blob/master/courses/models.py), 
+minus a few fields that are not relevant for this app.
+
+`Program`
+- title
+- description
+- thumbnail (image)
+- live (boolean - indicates whether the program is live in edX, et. al.)
+- source (string/enum - indicates where this was loaded/created from)
+
+`Course`
+- program (`Program` foreign key, **nullable** since a course doesn't necessarily belong to a program)
+- position_in_program (integer - indicates course sort order in a program)
+- title
+- description
+- thumbnail (image)
+- live (same as above)
+- source (same as above)
+
+`CourseRun`
+- course (`Course` foreign key)
+- external_id (string - the ID for the course run as it exists in
+  the courseware platform, e.g.: an edX course key like `course-v1:MITxPRO+AMx+1T2019`)
+- external_url (string - the URL for the course on the courseware platform)
+- title
+- start_date
+- end_date
+- enrollment_start
+- enrollment_end
+- live (same as above)
+
+Some things to debate/pay particular attention to:
+
+1. **`CourseRun.external_url`**
+    
+    We need to be able to link directly to a course on its courseware
+    platform. One option is to store the external id and, in our app logic,
+    generate the link to the course based on that id and the platform it
+    exists on (indicated by `source`). Another option is to simply store
+    the external url alongside the external id. That involves some redundant
+    data (the URL will almost certainly contain the external id), but the 
+    external id is highly unlikely to change and we will gain some 
+    flexibility, safety, and clarity by storing the URL separately.
+
+1. The `live` flag
+    
+    This is being added because we will want to be able to author some 
+    information for programs/courses that don't yet exist on the courseware 
+    platform.
+
+### Loading Course Data
+
+For the time being, the process of reflecting external course data in our
+models can be broken down into 2 parts:
+1. Loading legacy course data
+1. Periodically updating course data from our Open edX instance
+
+##### 1) Loading legacy course data
+
+We will likely receive legacy course data in the form of a data dump. The format 
+for that dump is not yet known. 
+
+**Proposals:**
+1. Create a management command that takes a file and creates/updates our 
+  course data.
+1. Ensure that the command is idempotent in case it has to be run
```

Comment:
We should consider what "idempotent" means in this case. If records exist in older datasets but don't exist in newer ones should they be removed? Don't have to answer questions like that now, but worth starting to think about how we define that for this operation. We could then retry an import by simply deleting that record and rerunning the sync.